pkg/packetbrokeragent/grpc_gspba.go

```diff
@@ -158,7 +158,19 @@ func (s *gsPbaServer) UpdateGateway(ctx context.Context, req *ttnpb.UpdatePacket
 	}
 
 	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "contact_info") {
+		adminUsrId := req.Gateway.GetAdministrativeContact().GetUserIds()
+		techUsrId := req.Gateway.GetTechnicalContact().GetUserIds()
 		adminContact, techContact := toPBContactInfo(req.Gateway.GetContactInfo())
+
+		// Prioritize administrative and technical contacts attached to the gateway instead of the ones present in
+		// ContactInfo slice.
+		if req.Gateway.AdministrativeContact != nil {
+			adminContact = &packetbroker.ContactInfo{Name: adminUsrId.GetUserId(), Email: adminUsrId.GetEmail()}
```

Comment:
The `email` in the `UserIdentifiers` message is one of our API foot guns.

It exists solely for the `forget-password` functionality, and it is never used in other places. In fact, noteven the forgotten password functionality uses it, if you check - `bunstore.GetUser` ignores it, and the implementation for `createTemporaryPassword` doesn't care about it either. I wish I could erase it from the API to be honest, but that's out of scope for this PR.

Back to this PR - you will need to retrieve the email via an RPC to the Identity Server in order to actually access the email. You will need to do this using cluster rights, and you will need to use the same tactic in order to retrieve the organization for the issue I've mentioned above (contact being an organization).

Reply:
Right. I think this is the tricky part of addressing this issue in PBA. Let us know if you need help on this @nicholaspcr.